enterprise/backend/src/metabase_enterprise/semantic_search/scoring.clj

```diff
@@ -14,6 +19,41 @@
   (let [prefix #(keyword (str %1 (name %2)))]
     [:coalesce (prefix "v." col-name) (prefix "t." col-name)]))
 
+(defn- view-count-percentile-query
+  [index-table p-value]
+  (let [expr [:raw "percentile_cont(" [:lift p-value] ") WITHIN GROUP (ORDER BY view_count)"]]
+    {:select   [:search_index.model [expr :vcp]]
+     :from     [[(keyword index-table) :search_index]]
+     :group-by [:search_index.model]
+     :having   [:is-not expr nil]}))
+
+(defn- view-count-percentiles*
+  [db index-table p-value]
+  (into {} (for [{:keys [model vcp]}
+                 (jdbc/execute! db
+                                (-> (view-count-percentile-query
+                                     index-table
+                                     p-value)
+                                    (sql/format {:quoted true}))
+                                {:builder-fn jdbc.rs/as-unqualified-lower-maps})]
+             [(keyword model) vcp])))
+
+(def ^{:private true
+       :arglists '([db index-table p-value])}
+  view-count-percentiles
+  (if config/is-prod?
+    (memoize/ttl view-count-percentiles*
```

Comment:
sticking `db` in a memoize table makes me a bit uneasy. I guess for now it's fine as it'd be a singleton datasource, but if we ever end up passing a Connection we may have object leaks which could grow even in 1h

Reply:
I was a bit worried about this as well tbh. What do you think about ignoring the db arg for caching purposes since it "should" never change? Obviously, that has bad edge cases of its own if it *does* change. Maybe `db` shouldn't be an argument to this function and it should just directly use `get-pgvector-datasource!` to enforce using the singleton connection?

Reply:
yeah I think that's better